python/cuml/cuml/manifold/umap.pyx

```diff
@@ -853,6 +857,39 @@ class UMAP(UniversalBase,
         del X_m
         return embedding
 
+    @property
+    def _n_neighbors(self):
+        return self.n_neighbors
+
+    @property
+    def _a(self):
+        return self.a
+
+    @property
+    def _b(self):
+        return self.b
+
+    @property
+    def _initial_alpha(self):
+        return self.learning_rate
+
+    @property
+    def _disconnection_distance(self):
+        return DISCONNECTION_DISTANCES.get(self.metric, np.inf)
```

Comment:
Aren't setters required to transfer attributes from umaplib to cuML?

Reply:
Hmm, I'm not sure how it was working before then. We never had matching name conventions with them, look at `_a` and `a` for example. Is it possible we did not support cpu train/gpu infer before?

Reply:
Actually, I think they are transfered during instantiation through hyperparameters listed in `get_param_names`. I do not know if we want to make these attributes writable.

Reply:
Okay, so we'll skip the setters then?